_observing-your-data/alerting/triggers.md

```diff
@@ -0,0 +1,142 @@
+
+### Sample scripts
+
+These scripts are Painless, not Groovy, but calling them Groovy in Jekyll gets us syntax highlighting in the generated HTML.
+
+```groovy
+// Evaluates to true if the query returned any documents
+ctx.results[0].hits.total.value > 0
+```
+
+```groovy
+// Returns true if the avg_cpu aggregation exceeds 90
+if (ctx.results[0].aggregations.avg_cpu.value > 90) {
+  return true;
+}
+```
+
+```groovy
+// Performs some crude custom scoring and returns true if that score exceeds a certain value
+int score = 0;
+for (int i = 0; i < ctx.results[0].hits.hits.length; i++) {
+  // Weighs 500 errors 10 times as heavily as 503 errors
+  if (ctx.results[0].hits.hits[i]._source.http_status_code == "500") {
+    score += 10;
+  } else if (ctx.results[0].hits.hits[i]._source.http_status_code == "503") {
+    score += 1;
+  }
+}
+if (score > 99) {
+  return true;
+} else {
+  return false;
+}
+```
+
+#### Trigger variables
+
+Variable | Data type | Description
+:--- | :--- | : ---
+`ctx.trigger.id` | String | The trigger ID.
+`ctx.trigger.name` | String | The trigger name.
+`ctx.trigger.severity` | String | The trigger severity.
+`ctx.trigger.condition`| Object | Contains the Painless script used when creating the monitor.
+`ctx.trigger.condition.script.source` | String | The language used to define the script. Must be Painless.
+`ctx.trigger.condition.script.lang` | String | The script used to define the trigger.
+`ctx.trigger.actions`| Array | An array with one element that contains information about the action the monitor needs to trigger.
+
+#### Other variables
+
+Variable | Data type | Description
+:--- | :--- : :---
+`ctx.results` | Array | An array with one element (`ctx.results[0]`). Contains the query results. This variable is empty if the trigger was unable to retrieve results. See `ctx.error`.
+`ctx.last_update_time` | Milliseconds | Unix epoch time of when the monitor was last updated.
+`ctx.periodStart` | String | Unix timestamp for the beginning of the period during which the alert triggered. For example, if a monitor runs every 10 minutes, a period might begin at 10:40 and end at 10:50.
+`ctx.periodEnd` | String | The end of the period during which the alert triggered.
+`ctx.error` | String | The error message if the trigger was unable to retrieve results or unable to evaluate the trigger, typically due to a compile error or null pointer exception. Null otherwise.
+`ctx.alert` | Object | The current, active alert (if it exists). Includes `ctx.alert.id`, `ctx.alert.version`, and `ctx.alert.isAcknowledged`. Null if no alert is active. Only available with query-level monitors.
+`ctx.dedupedAlerts` | Object | Alerts that have been triggered. OpenSearch keeps the existing alert to prevent the plugin from creating endless amounts of the same alerts. Only available with bucket-level monitors.
```

Comment:
```suggestion
`ctx.dedupedAlerts` | Object | Alerts that have been triggered. OpenSearch keeps the existing alert to prevent the plugin from creating endless numbers of the same alert. Only available with bucket-level monitors.
```